src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -434,55 +438,98 @@ def _resolve_service_principal(client, identifier):
     except ValueError:
         raise CLIError("service principal '{}' doesn't exist".format(identifier))
 
-def create_service_principal_for_rbac(name=None, password=None, years=1, #pylint:disable=too-many-arguments
-                                      scopes=None, role=None, expanded_view=None):
-    '''create a service principal that can access or modify resources
+def create_service_principal_for_rbac(name=None, password=None, years=1, #pylint:disable=too-many-arguments,too-many-statements,too-many-locals
+                                      scopes=None, role='Contributor', expanded_view=None):
+    '''create a service principal and configure its access to Azure resources
     :param str name: an unique uri. If missing, the command will generate one.
     :param str password: the password used to login. If missing, command will generate one.
     :param str years: Years the password will be valid.
     :param str scopes: space separated scopes the service principal's role assignment applies to.
-    :param str role: role the service principal has on the resources. only use with 'resource-ids'.
+           Defaults to the root of the current subscription.
+    :param str role: role the service principal has on the resources.
     '''
-    if bool(scopes) != bool(role):
-        raise CLIError("'--scopes' and '--role' must be used together.")
-    client = _graph_client_factory()
-    start_date = datetime.datetime.utcnow()
-    app_display_name = 'azure-cli-' + start_date.strftime('%Y-%m-%d-%H-%M-%S')
-    if name is None:
-        name = 'http://' + app_display_name # just a valid uri, no need to exist
-
-    end_date = start_date + relativedelta(years=years)
-    password = password or str(uuid.uuid4())
-    aad_application = create_application(client.applications, display_name=app_display_name, #pylint: disable=too-many-function-args
-                                         homepage='http://'+app_display_name,
-                                         identifier_uris=[name],
-                                         available_to_other_tenants=False,
-                                         password=password,
-                                         start_date=start_date,
-                                         end_date=end_date)
-    #pylint: disable=no-member
-    aad_sp = _create_service_principal(aad_application.app_id, bool(scopes))
-    oid = aad_sp.output.object_id if scopes else aad_sp.object_id
-
-    if scopes:
-        #It is possible the SP has not been propagated to all servers, so creating assignments
-        #might fail. The reliable workaround is to call out the server where creation occurred.
-        #pylint: disable=protected-access
+    import time
+    graph_client = _graph_client_factory()
+    role_client = _auth_client_factory().role_assignments
+    role = role or 'contributor'
```

Comment:
Casing doesn't matter, but yesm this code is redundant as we have a default set at the top